File: U2-Toolkit/.NET Samples/samples/VB.NET/UniData/EntityFramework_CodeFirst2/Student.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity

Namespace EntityFramework_CodeFirst2
    Public Class Student
        Sub New()
        End Sub


        <Key()> _
        Public Property ID() As Integer

        Public Property FNAME() As String
        Public Property LNAME() As String
        Public Property MAJOR() As String
        Public Property MINOR() As String
    End Class

    Public Class StudentContext
        Inherits DbContext
        Public Sub New()
        End Sub

        Public Property Customers() As DbSet(Of Student)
        Protected Overrides Sub OnModelCreating(modelBuilder As DbModelBuilder)

            modelBuilder.Entity(Of Student)().[Property](Function(s) s.ID).IsRequired()
        End Sub

    End Class

End Namespace

File: U2/Demos/U2-Toolkit/.NET Samples/samples/C#/UniData/EntityFramework_CodeFirst2/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity;

namespace EntityFramework_CodeFirst2
{
    public class Student
    {
        Student()
        {
        }

        [Key]
        public int ID { get; set; }

        public string FNAME { get; set; }
        public string LNAME { get; set; }
        public string MAJOR { get; set; }
        public string MINOR { get; set; }
    }

    public class StudentContext : DbContext
    {
        public StudentContext()
        {
        }
        public DbSet<Student> Customers { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Student>()
                .Property(s => s.ID)

                .IsRequired();
        }
    }




}
